service/resource/legacyv2/adapter/instance.go

```diff
@@ -0,0 +1,116 @@
+package adapter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/keyv2"
+)
+
+// template related to this adapter: service/templates/cloudformation/instance.yaml
+
+type instanceAdapter struct {
+	MasterAZ                     string
+	MasterIAMInstanceProfileName string
+	MasterImageID                string
+	MasterInstanceType           string
+	MasterSecurityGroupID        string
+	MasterSmallCloudConfig       string
+	MasterSubnetID               string
+}
+
+func (i *instanceAdapter) getInstance(customObject v1alpha1.AWSConfig, clients Clients) error {
+	if len(customObject.Spec.AWS.Masters) == 0 {
+		return microerror.Mask(invalidConfigError)
+	}
+
+	i.MasterAZ = keyv2.AvailabilityZone(customObject)
+	i.MasterImageID = keyv2.MasterImageID(customObject)
+	i.MasterInstanceType = keyv2.MasterInstanceType(customObject)
+	i.MasterIAMInstanceProfileName = keyv2.InstanceProfileName(customObject, prefixMaster)
+
+	// security group
+	// TODO: remove this code once the security group is created by cloudformation
+	// and add a reference in the template
+	groupName := keyv2.SecurityGroupName(customObject, prefixWorker)
+	describeSgInput := &ec2.DescribeSecurityGroupsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String(subnetDescription),
+				Values: []*string{
+					aws.String(groupName),
+				},
+			},
+			{
+				Name: aws.String(subnetGroupName),
+				Values: []*string{
+					aws.String(groupName),
+				},
+			},
+		},
+	}
+	output, err := clients.EC2.DescribeSecurityGroups(describeSgInput)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	if len(output.SecurityGroups) > 1 {
+		return microerror.Mask(tooManyResultsError)
+	}
+	i.MasterSecurityGroupID = *output.SecurityGroups[0].GroupId
+
+	// subnet ID
+	// TODO: remove this code once the subnet is created by cloudformation and add a
+	// reference in the template
+	subnetName := keyv2.SubnetName(customObject, suffixPrivate)
+	describeSubnetInput := &ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String(fmt.Sprintf("tag:%s", tagKeyName)),
+				Values: []*string{
+					aws.String(subnetName),
+				},
+			},
+		},
+	}
+	subnetOutput, err := clients.EC2.DescribeSubnets(describeSubnetInput)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	if len(subnetOutput.Subnets) > 1 {
+		return microerror.Mask(tooManyResultsError)
+	}
+	i.MasterSubnetID = *subnetOutput.Subnets[0].SubnetId
+
+	resp, err := clients.IAM.GetUser(&iam.GetUserInput{})
```

Comment:
We have `AccountID` for this now https://github.com/giantswarm/aws-operator/blob/master/service/resource/legacyv2/adapter/iam.go#L11

Reply:
Done, thanks that's much nicer.